File: src/main/java/net/dblsaiko/illuminate/client/LightSource.java
package net.dblsaiko.illuminate.client;

import net.dblsaiko.illuminate.Illuminate;
import net.dblsaiko.illuminate.client.api.Light;
import net.minecraft.entity.Entity;
import net.minecraft.nbt.NbtCompound;
import net.minecraft.world.World;
import org.joml.Vector3fc;

public class LightSource extends Entity {
    private final Light light;

    public LightSource(World world, Light light) {
        super(Illuminate.instance().entityTypes.lightSourceType(), world);
        this.light = light;

        Vector3fc lightPos = light.pos();
        this.setPosition(lightPos.x(), lightPos.y(), lightPos.z());
        this.setRotation(light.yaw(), light.pitch());

        this.prevX = this.getX();
        this.prevY = this.getY();
        this.prevZ = this.getZ();
        this.prevPitch = this.getPitch();
        this.prevYaw = this.getYaw();
        this.lastRenderX = this.getX();
        this.lastRenderY = this.getX();
        this.lastRenderZ = this.getZ();

    }

    public Light getLight() {
        return light;
    }

    @Override
    protected void initDataTracker() {

    }

    @Override
    protected void readCustomDataFromNbt(NbtCompound nbt) {

    }

    @Override
    protected void writeCustomDataToNbt(NbtCompound nbt) {

    }
}

File: src/main/kotlin/client/render/LightSource.kt
package therealfarfetchd.illuminate.client.render

import net.dblsaiko.illuminate.client.api.Light
import net.minecraft.entity.Entity
import net.minecraft.nbt.NbtCompound
import net.minecraft.network.Packet
import net.minecraft.network.listener.ClientPlayPacketListener
import net.minecraft.world.World
import therealfarfetchd.illuminate.Illuminate

class LightSource(world: World, light: Light) : Entity(Illuminate.LightEntityType, world) {

  init {
    setPosition(light.pos().x().toDouble(), light.pos().y().toDouble(), light.pos().z().toDouble())
    setRotation(light.yaw(), light.pitch())

    this.prevX = this.x
    this.prevY = this.y
    this.prevZ = this.z
    this.prevPitch = this.pitch
    this.prevYaw = this.yaw
    this.lastRenderX = this.x
    this.lastRenderY = this.y
    this.lastRenderZ = this.z
  }

  override fun initDataTracker() {}

  override fun readCustomDataFromNbt(nbt: NbtCompound?) {}

  override fun writeCustomDataToNbt(nbt: NbtCompound?) {}

  override fun createSpawnPacket(): Packet<ClientPlayPacketListener>? = null
}
